src/frontend/screens/ManualGpsScreen/UtmForm.tsx

```diff
@@ -38,98 +32,86 @@ const m = defineMessages({
-  const [{ value }, { updateDraft }] = useDraftObservation();
-
-  const [zoneNum, setZoneNum] = useState<string>(() => {
-    console.log("Setting initial state");
-    if (!location.savedPosition) return "";
-    try {
-      const { latitude, longitude } = location.savedPosition.coords;
-      const { zoneNum } = fromLatLon(latitude, longitude);
-      return zoneNum + "";
-    } catch (e) {
+const UtmForm = ({ coords, onValueUpdate }: FormProps) => {
+  const [zoneNum, setZoneNum] = React.useState(() => {
+    if (typeof coords?.lat === "number" && typeof coords?.lon === "number") {
+      try {
+        const { zoneNum } = fromLatLon(coords.lat, coords.lon);
+        return zoneNum + "";
+      } catch (e) {
+        return "";
+      }
+    } else {
       return "";
     }
   });
-  const [zoneLetter, setZoneLetter] = useState<string>(() => {
-    console.log("Setting initial state");
-    if (!location.savedPosition) return "";
-    try {
-      const { latitude, longitude } = location.savedPosition.coords;
-      const { zoneLetter } = fromLatLon(latitude, longitude);
-      return zoneLetter;
-    } catch (e) {
+
+  const [zoneLetter, setZoneLetter] = React.useState<string>(() => {
+    if (typeof coords?.lat === "number" && typeof coords?.lon === "number") {
+      try {
+        const { zoneLetter } = fromLatLon(coords.lat, coords.lon);
+        return zoneLetter;
+      } catch (e) {
+        return "";
+      }
+    } else {
       return "";
     }
   });
-  const [easting, setEasting] = useState<string>("");
-  const [northing, setNorthing] = useState<string>("");
 
-  React.useEffect(() => {
-    function handleSavePress() {
+  const [easting, setEasting] = React.useState(() => {
+    if (typeof coords?.lat === "number" && typeof coords?.lon === "number") {
       try {
-        const locationData = toLatLon({
-          zoneNum,
-          zoneLetter,
-          easting,
-          northing,
-        });
-        updateDraft({
-          lat: locationData.latitude,
-          lon: locationData.longitude,
-          metadata: {
-            ...(value || {}).metadata,
-            manualLocation: true,
-          },
-          tags: (value || {}).tags,
-        });
-        navigation.pop();
-      } catch (err) {
-        ToastAndroid.showWithGravity(
-          err.message,
-          ToastAndroid.LONG,
-          ToastAndroid.TOP
-        );
+        const { easting } = fromLatLon(coords.lat, coords.lon);
+        // TODO: Is it okay to round this to 3 decimal places?
+        return easting.toFixed(3);
+      } catch (e) {
+        return "";
       }
+    } else {
+      return "";
     }
+  });
+  const [northing, setNorthing] = React.useState(() => {
+    if (typeof coords?.lat === "number" && typeof coords?.lon === "number") {
+      try {
+        const { northing } = fromLatLon(coords.lat, coords.lon);
+        // TODO: Is it okay to round this to 3 decimal places?
+        return northing.toFixed(3);
```

Comment:
same question about precision here as [this comment](https://github.com/digidem/mapeo-mobile/pull/633/files#r671519224)